src/erlimem_session.erl

```diff
@@ -344,23 +344,27 @@ handle_info({_Ref,{StmtRef,Result}}, #state{stmts=Stmts}=State) when is_pid(Stmt
             ?Error("statement ~p not found in ~p", [StmtRef, [S|| {S,_} <- Stmts]]),
             {noreply, State}
     end;
-handle_info({{P,_} = From,Resp}, #state{stmts=Stmts}=State) when is_pid(P) ->
+handle_info({{P,_} = From, Resp}, #state{stmts=Stmts}=State) when is_pid(P) ->
     case Resp of
-            {error, Exception} ->
-                ?Debug("to ~p throw~n~p~n", [From, Exception]),
-                gen_server:reply(From,  {error, Exception}),
-                {noreply, State};
-            {ok, #stmtResult{stmtRef  = StmtRef} = SRslt} ->
-                ?Debug("RX ~p", [SRslt]),
-                %Rslt = {ok, SRslt, {?MODULE, StmtRef, self()}},
-                Rslt = {ok, SRslt},
-                ?Debug("statement ~p stored in ~p", [StmtRef, [S|| {S,_} <- Stmts]]),
-                gen_server:reply(From, Rslt),
-                {noreply, State#state{stmts=Stmts}};
-            Term ->
-                ?Debug("Async __RX__ ~p For ~p", [Term, From]),
-                gen_server:reply(From, Term),
-                {noreply, State}
+        {reply, {error, Exception}} ->
+            ?Debug("to ~p throw~n~p~n", [From, Exception]),
+            gen_server:reply(From,  {error, Exception}),
+            {noreply, State};
+        {reply, {ok, #stmtResult{stmtRef  = StmtRef} = SRslt}} ->
+            ?Debug("RX ~p", [SRslt]),
+            %Rslt = {ok, SRslt, {?MODULE, StmtRef, self()}},
+            Rslt = {ok, SRslt},
+            ?Debug("statement ~p stored in ~p", [StmtRef, [S|| {S,_} <- Stmts]]),
+            gen_server:reply(From, Rslt),
+            {noreply, State#state{stmts=Stmts}};
+        {reply, Term} ->
+            ?Debug("Sync __RX__ ~p For ~p", [Term, From]),
+            gen_server:reply(From, Term),
+            {noreply, State};
+        Resp ->
+            ?Debug("Async __RX__ ~p For ~p", [Resp, From]),
+            P ! Resp,
+            {noreply, State}
```

Comment:
But still we don't need the error clause unless we are doing something special with it, if we are just sending it back, we could just leave it to the last clause.